src/components/AssetIcon.tsx

```diff
@@ -51,12 +51,28 @@ const AssetWithNetwork: React.FC<AssetWithNetworkProps> = ({ assetId, icon, ...r
 }
 
 export const AssetIcon = ({ asset, assetId, src, ...rest }: AssetIconProps) => {
+  const _asset = useAppSelector(state => selectAssetById(state, assetId ?? ''))
```

Comment:
???

Reply:
`<AssetIcon /> currently takes in a maybe asset as props, so this ensures we don't bork things, see current consumption:

<img width="849" alt="image" src="https://user-images.githubusercontent.com/17035424/228640901-76618a60-1b38-4bdc-901f-2583d275941e.png">

I reckon we remove `asset` as a dependency altogether in a follow-up in profit of `assetId` now that we properly select it from store, though we'll have to make sure this doesn't bring any regressions

Reply:
Also removed the current obscure `asset` / `assetId` logic while at it and while regression testing this, added a `showNetworkIcon?: boolean` prop: 116ba00c03911343d05e0e9f98b9af1eb92f52cf

Turns out this logic served the purpose of showing the network icon vs. the asset icon